python/tskit/combinatorics.py

```diff
@@ -720,6 +721,207 @@ def is_symmetrical(self):
+    tmp_lab = [parent] + intermediate_nodes
+    assert len(edge_choice) == len(children) - 1
+    for node_lab, child_id, target_edge_id in zip(tmp_lab, children[1:], edge_choice):
+        target_edge = edges[target_edge_id]
+        # Insert in the right place, to keep edges in parent time order
+        edges.insert(target_edge_id, [node_lab, child_id])
+        edges.insert(target_edge_id, [node_lab, target_edge[1]])
+        target_edge[1] = node_lab
+    top_edge = edges.pop()  # remove the edge above the top node
+    assert top_edge[0] is None
+
+    # Re-map the internal nodes IDs so they are used in time order
+    real_node = iter(intermediate_nodes)
+    node_map = {c: c for c in children}
+    node_map[edges[-1][0]] = parent  # last edge == oldest parent
+    for e in reversed(edges):
+        # Reversing along the edges, parents are in inverse time order
+        for idx in (0, 1):  # look at parent (0) then child (1)
+            if e[idx] not in node_map:
+                node_map[e[idx]] = next(real_node)
+            e[idx] = node_map[e[idx]]
+    assert len(node_map) == len(intermediate_nodes) + len(children) + 1
+    return edges
+
+
+def split_polytomies(
+    ts,
+    *,
+    epsilon=None,
+    method=None,
+    record_provenance=True,
+    random_seed=None,
+):
+    """
+    Return a new tree sequence where extra nodes and edges have been inserted
+    so that any any node ``u`` with greater than 2 children (i.e. a multifurcation
+    or "polytomy") is resolved into successive bifurcations. Where ``method`` is
+    "random" the newly generated bifurcating topologies will be produced equiprobably,
+    by successive addition of edges to an initially empty list.
+
+    For further documentation, please refer to the :meth:`TreeSequence.split_polytomies`
+    method, which is the usual route through which this function is called.
+    """
+    allowed_methods = ["random"]
+    if epsilon is None:
+        epsilon = 1e-10
+    if method is None:
+        method = "random"
+    if method not in allowed_methods:
+        raise ValueError(f"Method must be chosen from {allowed_methods}")
+    rng = np.random.default_rng(seed=random_seed)
+    tables = ts.dump_tables()
+    # Store existing table data before clearing
+    existing_edges_left = (
+        tables.edges.left.copy()
+    )  # Will be changed if the edge is split
+    existing_edges_right = tables.edges.right.copy()  # Read only copy for efficiency
+    existing_edges_parent = tables.edges.parent.copy()  # Read only copy for efficiency
+    existing_edges_child = tables.edges.child.copy()  # Read only copy for efficiency
+    node_time = tables.nodes.time
+    tables.edges.clear()
+
+    edges_from_node = collections.defaultdict(set)  # Active descendant edge ids
+    nodes_changed = set()
+
+    for interval, e_out, e_in in ts.edge_diffs(include_terminal=True):
+        for edge in itertools.chain(e_out, e_in):
+            if edge.parent != tskit.NULL:
```

Comment:
``edge.parent`` can't be NULL if these are unmodified edges from the tree sequences

Same below

Reply:
Oh, good point. I always think that the root has a edge with a null parent, but that's not how we do it.